### PR TITLE
Download data update

### LINKED ISSUE #239: Ensure download_all_data works for new generators
Given the new function calls, we'll want to make sure we have a working way for users to download all datasets to disk.

### PR DESCRIPTION
**Pull request by ng390**
Closes #239

### CODE REVIEW

File: armory/data/datasets.py [hunk 1]
```diff
@@ -678,19 +679,52 @@ def _cache_dataset(dataset_dir: str, name: str, subpath: str):
...
     """
-    Download all datasets to cache.
+    Download all datasets for a scenario or requested datset to cache.
     """
-    errors = []
-    for name, func in SUPPORTED_DATASETS.items():
-        logger.info(f"Downloading (if necessary) dataset {name}")
-        try:
-            func()
-        except Exception:
```
Comment (davidslater) [inline armory/data/datasets.py]: We may want (for testing purposes) to be able to call this with no arguments, and have it actually go and download all datasets.

File: armory/data/datasets.py [hunk 2]
```diff
@@ -678,19 +679,52 @@ def _cache_dataset(dataset_dir: str, name: str, subpath: str):
...
+        raise ValueError(
+            f"dataset {dataset_name} not supported. Must be one of {list(SUPPORTED_DATASETS.keys())}"
+        )
+
+    func = SUPPORTED_DATASETS[dataset_name]
+
+    logger.info(f"Downloading (if necessary) dataset {dataset_name}...")
+
+    if dataset_name == "imagenet_adversarial":
+        kwargs = {"epochs": 1, "batch_size": 1}
```
Comment (davidslater) [inline armory/data/datasets.py]: "clean" or "adversarial" can be the split_type for imagenet_adversarial

Comment (davidslater) [inline armory/data/datasets.py]: If there were kwarg defaults for all of the arguments for the dataset functions (split_type="train" or "clean", epochs=1, batch_size=1), then we wouldn't need to specify here - we could just call the dataset function to download.

**Comment by ng390** [comment]
Ready for review, though Issue #380 (as a separate PR) may greatly improver user experience.

File: armory/scenarios/download_configs/scenarios-set1.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        },
+        "resisc45-image-classification": {
+            "dataset_name": [
+                "resisc45"
+            ],
+            "weights_file": [
+                "densenet121-resisc45-val_acc-0.93.hdf5"
+            ]
+        },
+        "traffic-sign-backdoor-poisoning": {
```
Comment (seanpmorgan) [inline armory/scenarios/download_configs/scenarios-set1.json:28]:
```suggestion
        "german-traffic-sign-image-poisoning": {
```

**Comment by davidslater** [comment]
Is there a way we can suppress the warning:
/opt/conda/lib/python3.7/site-packages/urllib3/connectionpool.py:847: InsecureRequestWarning: Unverified HTTPS request is being made. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecureRequestWarning)
??

It seems to be made every few seconds.

**Comment by davidslater** [comment]
I'm getting errors running this on galileo:
(armory-base)davidslater@galileo:~/armory$ python -m armory download scenarios-set1            
2020-04-14 13:59:22 galileo armory.paths[18991] INFO Creating armory directories if they do not exist
Downloading requested datasets and model weights...                                                                                                                                            
2020-04-14 13:59:23 galileo armory.docker.management[18991] INFO ARMORY Instance 27eefdd095 created.
2020-04-14 13:59:27 27eefdd09598 armory.data.datasets[7] INFO Downloading (if necessary) dataset cifar10...
                                                                                               
2020-04-14 13:59:27 27eefdd09598 armory.data.datasets[7] ERROR Loading dataset cifar10 failed.                           
Traceback (most recent call last):                                                             
  File "/workspace/armory/data/datasets.py", line 515, in _download_data                                    
    func()                                                                                     
  File "/workspace/armory/data/datasets.py", line 228, in cifar10                                         
    cache_dataset=cache_dataset,                                                               
  File "/workspace/armory/data/datasets.py", line 135, in _generator_from_tfds                 
    dataset_dir, dataset_name=dataset_name,                                                    
  File "/workspace/armory/data/datasets.py", line 437, in _cache_dataset                       
    name=name,                                                                                 
  File "/workspace/armory/data/utils.py", line 144, in download_verify_dataset_cache           
    os.makedirs(cache_dir, exist_ok=True)                                                      
  File "/opt/conda/lib/python3.7/os.py", line 211, in makedirs                                 
    makedirs(head, exist_ok=exist_ok)                                                          
  File "/opt/conda/lib/python3.7/os.py", line 221, in makedirs                                 
    mkdir(name, mode)                                                                          
PermissionError: [Errno 13] Permission denied: '/armory/datasets'

**Pull request merged**